### ISSUE TITLE
Couldn't add Numerics Converter as package from another package

### ISSUE THREAD

**Issue by psydack** [KuraiAndras/NumericsConverter#2]
Hello! 

[SUGESTION]
I couldn't add **NumericsConverter** to another package because packages can not use scoped registries (needed by upm). Maybe it will worth to make a repository to be able to add at unity packager as "git url".

Thanks for putting effort on this.

**Comment by KuraiAndras** [comment]
Did you try to add it using the openupm cli?

**Comment by psydack** [comment]
Yes, but openupm add a scoped registry to make manageable by Package Manager, but packages can't use it. 
Try to make a package that needs your package (com.kuraiandras.numericsconverter).

```json
{
  "name": "Something",
  "displayName": "New Package",
  "version": "0.0.1",
  "description": "Sample Description",
  "unity": "2020.3",
  "dependencies": {
    "com.kuraiandras.numericsconverter" : "2.0.0" 
  },
  "keywords": [
  ],
  "author": {
    "name": "Oi",
    "url": "https://github.com/KuraiAndras/NumericsConverter"
  }
}
```

**Comment by KuraiAndras** [comment]
In this case shouldn`t just adding the registry scoped to this project to the project consuming the libaray fix it?

**Comment by psydack** [comment]
No because the unity don't use, I don't know why, scoped registry on packages. I tried but failed.

**Comment by KuraiAndras** [comment]
Then you should be able to reference the project just with a git dependency like so (according to [documentation](https://docs.unity3d.com/Manual/upm-git.html)):
```
"com.kuraiandras.numericsconverter": "https://github.com/KuraiAndras/NumericsConverter.git?path=/NumericsConverter/Assets/NumericsConverter"
```

for a specific version under a tag this should work:

```
"com.kuraiandras.numericsconverter": "https://github.com/KuraiAndras/NumericsConverter.git?path=/NumericsConverter/Assets/NumericsConverter#2.0.0"
```

**Comment by KuraiAndras** [comment]
Does this work for you @psydack?

**Comment by psydack** [comment]
Yes. Thanks for the explanation.

**Comment by psydack** [comment]
Sorry to reopen. This will only works on main manifest, if you put this on dependency from a package that won't work. 

Make and empty folder (as a package) and put on packages folder
Put your repo as dependency from that package

![image](https://user-images.githubusercontent.com/961971/119580220-5d1f2b80-bd96-11eb-9ba6-06386b3cd9a6.png)

**Comment by KuraiAndras** [comment]
Can you show how you reference the package? Or a link to a full repro?

**Comment by psydack** [comment]
Sure: https://github.com/psydack/uimgui/commit/d2706f555578c8f79349d156453cbe38998b034b#diff-7ae45ad102eab3b6d7e7896acd08c427a9b25b346470d7bc6507b6481575d519

**Comment by KuraiAndras** [comment]
Did you try specifiying a specific version/tag?
```"com.kuraiandras.numericsconverter": "https://github.com/KuraiAndras/NumericsConverter.git?path=/NumericsConverter/Assets/NumericsConverter#2.0.1"```

**Comment by psydack** [comment]
Yes, same error: 
![image](https://user-images.githubusercontent.com/961971/119667959-1f111e80-be0d-11eb-9248-8742d4ef17c8.png)

**Comment by KuraiAndras** [comment]
https://issuetracker.unity3d.com/issues/package-manager-doesnt-resolve-package-dependencies-when-using-git-url-to-add-a-package-which-contains-a-git-url-dependency
According to this, it is by design. You can't add git dependencies to a package.json file.

Are you sure you can't make this work with the openupm package?
```
this in you package.json
"dependencies": {
    "com.kuraiandras.numericsconverter" : "2.0.1" 
  },
```

And then add the scoped registry from openupm in the consuming project?

**Comment by psydack** [comment]
I could add that as a need to the person who will use my package, but
that's too much for a simple conversion. I manually embed to my packages
your package, it's okay to me:
https://github.com/psydack/uimgui/commit/d2706f555578c8f79349d156453cbe38998b034b.
I am just leaving this issue because some can have this error too.
